src/main/java/aquality/selenium/core/waitings/ConditionalWait.java

```diff
@@ -0,0 +1,116 @@
+package aquality.selenium.core.waitings;
+
+import aquality.selenium.core.applications.IApplication;
+import aquality.selenium.core.configurations.ITimeoutConfiguration;
+import aquality.selenium.core.localization.ILocalizationManager;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BooleanSupplier;
+
+public class ConditionalWait implements IConditionalWait {
+
+    private IApplication application;
+    private ITimeoutConfiguration timeoutConfiguration;
+    private ILocalizationManager localizationManager;
+
+    @Inject
+    public ConditionalWait(Provider<IApplication> applicationProvider, ITimeoutConfiguration timeoutConfiguration, ILocalizationManager localizationManager) {
+        application = applicationProvider.get();
+        this.timeoutConfiguration = timeoutConfiguration;
+        this.localizationManager = localizationManager;
+    }
+
+    @Override
+    public boolean waitFor(BooleanSupplier condition, Long timeoutInSeconds, Long pollingIntervalInMilliseconds, String message, Collection<Class<? extends Throwable>> exceptionsToIgnore) {
+        try {
+            waitForTrue(condition, timeoutInSeconds, pollingIntervalInMilliseconds, message, exceptionsToIgnore);
+            return true;
+        } catch (TimeoutException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public void waitForTrue(BooleanSupplier condition, Long timeoutInSeconds, Long pollingIntervalInMilliseconds, String message, Collection<Class<? extends Throwable>> exceptionsToIgnore) throws TimeoutException {
+        if (condition == null) {
```

Comment:
let's use Optional.ofNullable(condition).orElseThrow(() -> new IllegalArgumentException("<what you want to say>")

and for nullable timeouts it could be
long timeout = Optional.ofNullable(timeoutInSeconds).orElseGet

Reply:
fixed